### PR TITLE
[MOL-14046][HFAH] auto uppercase IC input field for eWills & HDB carparking

### PR DESCRIPTION
**Pull request by Hamiriza**
**Changes**
Added  a new prop to cater for the auto uppercase IC input field

- delete branch
- You may refer to this [MOL-14046](https://jira.ship.gov.sg/browse/MOL-14046)

### CODE REVIEW

File: src/masked-input/types.ts [hunk 1]
```diff
@@ -13,6 +13,7 @@ export interface MaskedInputProps extends InputProps {
     onMask?: (() => void) | undefined;
     onUnmask?: (() => void) | undefined;
     disableMask?: boolean | undefined;
+    transformUppercase?: boolean;
```
Comment (keithtxw) [inline src/masked-input/types.ts:16]: Thinking for more flexibility, let's make it a string literal type
```
transform?: "uppercase" | "lowercase" | undefined;
```

**Review by keithtxw — CHANGES_REQUESTED** [review]
Do also document down this new prop in the stories too

File: src/masked-input/types.ts (hunk 1 above)
Comment (thyAshley) [reply src/masked-input/types.ts:16]: m just thinking if we need these transform, cause our nric validator fails any non-uppercase input, should uppercase be the default?

Comment (Hamiriza) [reply src/masked-input/types.ts:16]: I was thinking of the possibility of this component being reused for other purposes such as password field, so one way is to just pass the props later on, depending on the validator behaviour. I think going for flexibility would make more sense now?

File: src/masked-input/masked-input.tsx [hunk 2]
```diff
@@ -57,7 +58,15 @@ const Component = (
     };
 
     const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-        setUpdatedValue(event.target.value);
+        let value = event.target.value;
```
Comment (qroll) [inline src/masked-input/masked-input.tsx:61]: onChange still receives the original `event.target.value`, so you may want to assign the value on the event object

File: src/masked-input/types.ts (hunk 1 above)
Comment (keithtxw) [reply src/masked-input/types.ts]: yup agree. I am seeing this more than just for nric. The user for nric fields should set the prop to `uppercase`

File: stories/form/form-masked-input/props-table.tsx [hunk 3]
```diff
@@ -82,6 +82,12 @@ const DATA: ApiTableSectionProps[] = [
                 description: "The callback function when field is not masked",
                 propTypes: ["() => void"],
             },
+            {
+                name: "transformInput",
+                description:
+                    "Specifies whether the input is tranformed into uppercase or lowercase",
```
Comment (keithtxw) [inline stories/form/form-masked-input/props-table.tsx:88]:
```suggestion
                    "Specifies whether the input value is to be tranformed into uppercase or lowercase format",
```

File: stories/form/form-masked-input/form-masked-input.stories.mdx [hunk 4]
```diff
@@ -40,6 +40,16 @@ import { Form } from "@lifesg/react-design-system/form";
                     value="S1234567D"
                     maskRange={[2, 5]}
                 />
+                <Form.MaskedInput
+                    label="This is the mask field with uppercase input"
```
Comment (keithtxw) [inline stories/form/form-masked-input/form-masked-input.stories.mdx:44]:
```suggestion
                    label="This is set to be transformed to uppercase"
```

File: stories/form/form-masked-input/form-masked-input.stories.mdx [hunk 5]
```diff
@@ -40,6 +40,16 @@ import { Form } from "@lifesg/react-design-system/form";
...
+                    maskRange={[2, 5]}
+                    transformInput={"uppercase"}
+                />
+                <Form.MaskedInput
+                    label="This is the mask field with lowercase input"
```
Comment (keithtxw) [inline stories/form/form-masked-input/form-masked-input.stories.mdx:49]:
```suggestion
                    label="This is set to be transformed to lowercase"
```

**Review by keithtxw — CHANGES_REQUESTED** [review]

**Review by keithtxw — APPROVED** [review]

**Pull request merged**